### PR TITLE
chore(localization): updating strings for Settings page (resolves #931)

### LINKED ISSUE #931: Update text for Settings

### PR DESCRIPTION
**Pull request by chosww**
Explain what this PR does.

- Update string texts in settings pages
- Refactor DestroyUserRequest's condition check for deletable user to User model
- Add new modal view for removing members from org

### Prerequisites

If this PR changes PHP code or dependencies:

- [x] I've run `composer format` and fixed any code formatting issues.
- [x] I've run `composer analyze` and addressed any static analysis issues.
- [x] I've run `php artisan test` and ensured that all tests pass.
- [x] I've run `composer localize` to update localization source files and committed any changes.

If this PR changes CSS or JavaScript code or dependencies:

- [x] I've run `npm run lint` and fixed any linting issues.
- [ ] I've run `npm run build` and ensured that CSS and JavaScript assets can be compiled.

### CODE REVIEW

File: resources/views/settings/delete-account.blade.php [hunk 1]
```diff
@@ -6,26 +6,94 @@
...
+                {{ __('You cannot delete your account because you are the only administrator of your organization. Please make another member from your organization an administrator first.') }}
+            </x-hearth-alert>
+        </div>
+    @endif
+    <h2>{{ __('Are you sure you want to delete your account?') }}</h2>
+    <p>{{ __('If you delete your account:') }}</p>
+    @if ($user->context === 'individual')
+        <ul>
+            <li>
+                <p>{{ __('you will no longer be able to access information about the [number] engagements you are participating in') }}
```
Comment (greatislander) [inline resources/views/settings/delete-account.blade.php]: In all these instances, in order to let the translation work start we'll need to add an actual placeholder. So I'd do something like this as a temporary measure— we can fill in a real number later:

```suggestion
                <p>{{ __('you will no longer be able to access information about the :count engagements you are participating in', ['count' => 0]) }}
```

File: resources/views/components/remove-member-modal.blade.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<div x-data="modal()">
+    <button class="secondary" type="button" @click="showModal">{{ __('Remove') }}</button>
+    <template x-teleport="body">
+        <div class="modal-wrapper" x-show="showingModal">
+            <div class="modal stack" @keydown.escape.window="hideModal">
+                <h3>{{ __('Remove :member from your organization', ['member' => $member->name]) }}</h3>
+
+                <p>
+                    {{ __('Are you sure you want to remove :member from :organization? You cannot undo this.', ['member' => $member->name, 'organization' => $membershipable->name]) }}
```
Comment (greatislander) [inline resources/views/components/remove-member-modal.blade.php]: If the member is the current user, it this should say "Are you sure you want to leave :organization?".

File: resources/views/components/remove-member-modal.blade.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<div x-data="modal()">
+    <button class="secondary" type="button" @click="showModal">{{ __('Remove') }}</button>
```
Comment (greatislander) [inline resources/views/components/remove-member-modal.blade.php]: This should read "Leave" if the member is the current user.

**Review by greatislander — CHANGES_REQUESTED** [review]

File: resources/views/settings/delete-account.blade.php [hunk 4]
```diff
@@ -6,26 +6,94 @@
...
+            <x-hearth-alert type="warning">
+                {{ __('Once you delete your account, you will not be able to recover it.') }}
+            </x-hearth-alert>
+        </div>
+    @endif
 
-    <form action="{{ localized_route('users.destroy') }}" method="post" novalidate>
+    <!-- Form Validation Errors -->
+    @include('partials.validation-errors')
+    <form class="@if ($user->isOnlyAdministratorOfOrganization() || $user->isOnlyAdministratorOfRegulatedOrganization()) disabled @endif" action="{{ localized_route('users.destroy') }}"
```
Comment (chosww) [inline resources/views/settings/delete-account.blade.php:88]: @greatislander I know you suggested me to update class name in this manner over ternary operator, but this method adds extra white spaces in the class name; the class name becomes " disabled ", so the style in _form.css is not applying as expected.

File: resources/views/components/remove-member-modal.blade.php (hunk 3 above)
Comment (chosww) [reply resources/views/components/remove-member-modal.blade.php]: Should we also update the modal title to something like "Leave from your organization"?

File: resources/views/settings/delete-account.blade.php (hunk 4 above)
Comment (greatislander) [reply resources/views/settings/delete-account.blade.php:88]: If the class name is `" disabled"` it still should apply… there might be another issue.

Comment (chosww) [reply resources/views/settings/delete-account.blade.php:88]: oh, I see when I just hard coded class name to be "disabled", the style is till not applied.

**Review by greatislander — CHANGES_REQUESTED** [review]
Add `description()` methods to `TeamRole` and `EngagementRecruitment` and derive hints from them

File: resources/views/components/remove-member-modal.blade.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<div x-data="modal()">
+    <button class="secondary" type="button"
+        @click="showModal">{{ $member->id === $user->id ? __('Leave') : __('Remove') }}</button>
+    <template x-teleport="body">
+        <div class="modal-wrapper" x-show="showingModal">
+            <div class="modal stack" @keydown.escape.window="hideModal">
+                <h3>{{ __('Remove :member from your organization', ['member' => $member->name]) }}</h3>
```
Comment (greatislander) [inline resources/views/components/remove-member-modal.blade.php]:
```suggestion
                <h3>
                @if ($member->id === $user->id)
                    {{ __('Leave this organization') }}
                @else
                    {{ __('Remove :member from your organization', ['member' => $member->name]) }}
              	@endif
              	</h3>
```

**Review by greatislander — CHANGES_REQUESTED** [review]

**Review by greatislander — APPROVED** [review]

**Comment by cherylhjli** [comment]
Please refer to the colour coding on the latest wireframes for strings that are not matching the wires.

**Comment by greatislander** [comment]
@chosww ☝🏻

**Comment by cherylhjli** [comment]
Settings - blocked
- Block modal: There’s two lines that both say “They will not know you have blocked them.”

**Pull request merged**